### PR TITLE
zh-cn: update the translation of WebIDL information

### LINKED ISSUE #11063: zh-cn: complete translation for mdn guidelines
We have 3 sections that need to be done:

- [Community](https://developer.mozilla.org/zh-CN/docs/MDN/Community)
  What we could do for the community and what we have.

  - [x] [overview](https://developer.mozilla.org/en-US/docs/MDN/Community) #11348
  - [x] [Contributing to MDN](https://developer.mozilla.org/en-US/docs/MDN/Community/Contributing)
    - [x] [MDN Web Docs Repositories](https://developer.mozilla.org/en-US/docs/MDN/Community/Contributing/Our_repositories) #15724
    - [x] [Getting started with MDN Web Docs](https://developer.mozilla.org/en-US/docs/MDN/Community/Contributing/Getting_started) #15795
    - [x] [Security vulnerability response steps](https://developer.mozilla.org/en-US/docs/MDN/Community/Contributing/Security_vulnerability_response) #16151
    - [x] [MDN Web Docs Localization](https://developer.mozilla.org/en-US/docs/MDN/Community/Contributing/Translated_content) #15732
  - [x] [Open source etiquette](https://developer.mozilla.org/en-US/docs/MDN/Community/Open_source_etiquette)
  - [x] [Communication channels](https://developer.mozilla.org/en-US/docs/MDN/Community/Communication_channels) #13511
  - [x] [Community discussions](https://developer.mozilla.org/en-US/docs/MDN/Community/Discussions) #13815
  - [x] [Learn forum](https://developer.mozilla.org/en-US/docs/MDN/Community/Learn_forum)
  - [x] [Issues](https://developer.mozilla.org/en-US/docs/MDN/Community/Issues) #15706
  - [x] [PR](https://developer.mozilla.org/en-US/docs/MDN/Community/Pull_requests) #13953
  - [x] [MDN content](https://developer.mozilla.org/en-US/docs/MDN/Community/Mdn_content) #15049
  - [x] [Role and teams](https://developer.mozilla.org/en-US/docs/MDN/Community/Roles_teams) #15400

- [Writing guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines)
  How to do contributions

  - [x] [Experimental, deprecated, and obsolete](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete) #13406
  - [x] [overview](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines) #11064
  - [x] [What we write](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/What_we_write) #12043
    - [x] [Criteria for inclusion](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/What_we_write/Criteria_for_inclusion) #16456

  - [x] [Writing style](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide) #13289
    - [x] [Code style guide](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide) #16990
    - [x] [HTML guideline](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/HTML) #17021
    - [x] [CSS guideline](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/CSS) #17413
    - [x] [JavaScript guideline](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript) #17007
    - [x] [Shell prompt guideline](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/Shell) #17048

  - [x] [How to](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto) (many sub pages left here) #12296

    - [x] [How to write an API reference](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Write_an_api_reference) #12864
      - [x] [API reference sidebars](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Write_an_API_reference/Sidebars) #16512
      - [x] [Information contained in a WebIDL file](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Write_an_API_reference/Information_contained_in_a_WebIDL_file) #17654
    - [x] [How to create, move, delete, and edit pages](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Creating_moving_deleting) #13065
    - [x] [How to write in Markdown](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN) #15064
    - [x] [How to add images and media](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Images_media) #17559
    - [x] [How to document a CSS property](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Document_a_CSS_property) #15585
    - [x] [How to document an HTTP header](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Document_an_HTTP_header) #15392
    - [x] [How to document web errors](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Document_web_errors) #15616
    - [x] [How to research a technology](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Research_technology) #15780 by @yarusome
    - [x] [How to use structured data](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/JSON_Structured_data) #16506
    - [x] [How to write an entry in the glossary](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Write_a_new_entry_in_the_glossary) #10973

  - [x] [Page struct](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures) (many sub pages left here)
    - [x] [Banners and notices](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Banners_and_notices) #13048
    - [x] [Code example](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples) (already done)
    - [x] [BCD guide](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Compatibility_tables) #13584
    - [x] [Live sample](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Live_samples) (already done)
    - [x] [Page types](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types) #13876 #15395
    - [x] [Quicklinks](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Quicklinks) #13655
    - [x] [Specification tables](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Specification_tables) #13245
    - [x] [Syntax sections](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Syntax_sections) #15857
    - [x] [Using macros](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros) #13584
      - [x] [Commonly-used macros](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros/Commonly_used_macros) #15338
      - [x] [Other macros](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros/Other) (already done)
    - [ ] [Feature status](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Feature_status) #19492
  - [x] [copyright](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Attrib_copyright_license) (small effort) #13575

- [x] [sidebar translation](https://github.com/mdn/yari/blob/main/kumascript/macros/MDNSidebar.ejs)
mdn/yari#9129
- [ ] [changelog](https://developer.mozilla.org/en-US/docs/MDN/Changelog)

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of WebIDL information

### Related issues and pull requests

Fixes: #11063
Preparation: mdn/content#31522

### CODE REVIEW

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 1]
```diff
@@ -1,221 +1,262 @@
...
+l10n:
+  sourceCommit: c6497f6347a9dd6d6844b520b1ba8542552653a8
 ---
 
 {{MDNSidebar}}
 
 在编写有关 API 的文档时，信息来源很多：规范描述了应该实现的内容以及模型，实现描述了实际放在浏览器中的内容。WebIDL 文件是一种非常简洁的方式，可以提供很多（但不是全部）有关 API 的信息。本文档提供了有助于理解 WebIDL 语法的参考。
 
-IDL 代表**_接口定义语言_**，它用于描述 API。在更广泛的计算领域，有几种 IDL。在浏览器领域，我们使用的 IDL 称为*WebIDL*。有两种 WebIDL 可用：WebIDL 规范中给出的一种，以及在浏览器中实现的一种。规范是规范引用，浏览器 WebIDL 描述了在特定浏览器中实际实现的内容，并包含其他内容，例如注释和有关非标准元素的信息。
+IDL 代表**_接口定义语言_**，它用于描述 API。在更广泛的计算领域，有几种 IDL。在浏览器领域，我们使用的 IDL 被称为 _WebIDL_。有两种 WebIDL 可用：一种是在 WebIDL 规范中给出的，一种是在浏览器中实现的。规范是标准参考，而浏览器的 WebIDL 描述了在特定浏览器中实际实现的内容，并包含其他内容，例如注释、有关非标准元素的信息，以及对 IDL 规范的特定于浏览器的扩展。
```
Comment (awxiaoxian2020) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:12]:
```suggestion
IDL 代表**_接口定义语言_**，它用于描述 API。在更广泛的计算领域，IDL 有不同的种类。在浏览器领域，我们使用的 IDL 被称为 _WebIDL_。有两种 WebIDL 可用：一种是在 WebIDL 规范中给出的，一种是在浏览器中实现的。规范是标准参考，而浏览器的 WebIDL 描述了在特定浏览器中实际实现的内容，并包含其他内容，例如注释、有关非标准元素的信息，以及对 IDL 规范的特定于浏览器的扩展。
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 2]
```diff
@@ -1,221 +1,262 @@
...
-If the keyword `readonly` is present, the property can't be modified. It must be marked as read-only:
+如果关键字 `readonly` 存在，则该属性不能被修改。它必须标记为只读：
 
-- In the interface, by adding the \\{{ReadOnlyInline}} macro next to its definition term.
-- In the first sentence of its own page, by starting the description with: _The read-only **`HTMLMediaElement.error`** property…_
-- By adding the `Read-only` tag to its own page.
-- By starting its description in the interface page with _Returns…_
+- 在接口页面中，在第一句话中使用 \\{{ReadOnlyInline}} 宏。
+- 在其自己的页面中，在第一句话中使用中使用描述：_**`HTMLMediaElement.error`** 只读属性…_
+- 在其自己的页面中，添加 `Read-only` 标签。
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:228]: 现在应该不用页面Tag了

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 3]
```diff
@@ -1,221 +1,262 @@
...
 
-If the keyword `readonly` is present, the property can't be modified. It must be marked as read-only:
+如果关键字 `readonly` 存在，则该属性不能被修改。它必须标记为只读：
 
-- In the interface, by adding the \\{{ReadOnlyInline}} macro next to its definition term.
-- In the first sentence of its own page, by starting the description with: _The read-only **`HTMLMediaElement.error`** property…_
-- By adding the `Read-only` tag to its own page.
-- By starting its description in the interface page with _Returns…_
+- 在接口页面中，在第一句话中使用 \\{{ReadOnlyInline}} 宏。
+- 在其自己的页面中，在第一句话中使用中使用描述：_**`HTMLMediaElement.error`** 只读属性…_
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:227]:
```suggestion
- 在自己页面的第一句话中，用以下文字开始描述：_**`HTMLMediaElement.error`** 只读属性…_
```

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 4]
```diff
@@ -1,221 +1,262 @@
...
 ```
 
-If the keyword `readonly` is present, the property can't be modified. It must be marked as read-only:
+如果关键字 `readonly` 存在，则该属性不能被修改。它必须标记为只读：
 
-- In the interface, by adding the \\{{ReadOnlyInline}} macro next to its definition term.
-- In the first sentence of its own page, by starting the description with: _The read-only **`HTMLMediaElement.error`** property…_
-- By adding the `Read-only` tag to its own page.
-- By starting its description in the interface page with _Returns…_
+- 在接口页面中，在第一句话中使用 \\{{ReadOnlyInline}} 宏。
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:226]:
```suggestion
- 在接口页面中，在其定义项旁边添加 \\{{ReadOnlyInline}} 宏。
```

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 5]
```diff
@@ -224,272 +265,337 @@ partial interface Document {
...
+迭代器将同时迭代 _valueType_ 类型的值和 _keyType_ 类型的键，即值对组（value pair）。生成的方法有：
 
-- `entries()` that returns an {{jsxref('iterator')}} on the indexes (of type _keyType_). E.g. {{domxref('FormData.entries()')}}
-- `values()` that returns an {{jsxref('iterator')}} on the values. E.g. {{domxref('FormData.values()')}}
-- `keys()` that returns an {{jsxref('iterator')}} on the keys. E.g. {{domxref('FormData.keys()')}}
-- Once [Firefox bug 1216751](https://bugzil.la/1216751) lands, `forEach()`.
+- `entries()`，返回一个值对组[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('FormData.entries()')}}。
+- `values()`，返回一个值[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('FormData.values()')}}。
+- `keys()`，返回一个键[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('FormData.keys()')}}。
+- `forEach()`，会对列表中的每一个实体调用一次给定的构造函数。例如，请参见 {{domxref('Headers.forEach()')}}。
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:457]:
```suggestion
- `forEach()`，会对列表中的每一个实体调用一次给定的回调函数。例如，请参见 {{domxref('Headers.forEach()')}}。
```

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 6]
```diff
@@ -224,272 +265,337 @@ partial interface Document {
...
+迭代器将迭代类型为 _valueType_ 的值。生成的方法有：
 
-- `entries()`, which returns an {{jsxref('iterator')}} on the indexes (that are `unsigned long`).
-- `values()`, which returns an {{jsxref('iterator')}} on the values.
-- `keys()`, which returns an {{jsxref('iterator')}} on the keys, that are its indexes (that are `unsigned long`). In the case of value iterators, `keys()` and `entries()` are identical.
-- `forEach()`, which returns an {{jsxref('iterator')}} on the keys that calls a given callback function one for each entry in the list.
+- `entries()`，返回一个索引（`unsigned long` 类型）[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。
+- `values()`，返回一个值[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。
+- `keys()`，返回一个键（即，索引，`unsigned long` 类型）[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。对于值迭代器，`keys()` 和 `entries()` 是相同的。
+- `forEach()`，会对列表中的每一个实体调用一次给定的构造函数。
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:436]:
```suggestion
- `forEach()`，会对列表中的每一个实体调用一次给定的回调函数。
```

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 7]
```diff
@@ -224,272 +265,337 @@ partial interface Document {
...
 setlike<valueType>
 ```
 
-The generated properties will be:
+生成的属性有：
+
+- `entries()`，返回一个索引[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('NodeList.entries()')}}。
+- `values()`，返回一个值[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('NodeList.values()')}}。
+- `keys()`，返回一个键[迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)。例如，请参见 {{domxref('NodeList.keys()')}}。
+- `forEach()`，会对列表中的每一个实体调用一次给定的构造函数。例如，请参见 {{domxref('NodeList.forEach()')}}。
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:479]:
```suggestion
- `forEach()`，会对列表中的每一个实体调用一次给定的回调函数。例如，请参见 {{domxref('NodeList.forEach()')}}。
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
some suggestions

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md (hunk 2 above)
Comment (yin1999) [reply files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:228]:
```suggestion
```

File: files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md [hunk 8]
```diff
@@ -1,221 +1,261 @@
 ---
-title: Information contained in a WebIDL file
+title: WebIDL 文件中包含的信息
 slug: MDN/Writing_guidelines/Howto/Write_an_API_reference/Information_contained_in_a_WebIDL_file
+l10n:
+  sourceCommit: c6497f6347a9dd6d6844b520b1ba8542552653a8
```
Comment (jasonren0403) [inline files/zh-cn/mdn/writing_guidelines/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.md:5]:
```suggestion
  sourceCommit: 24b4448c9dbaf3cb9084ad69a85c72e6aa2f72f9
```

**Review by jasonren0403 — APPROVED** [review]
LGTM, Thanks @yin1999 and @awxiaoxian2020 👍

**Pull request merged**